### PR TITLE
LPD-34975

### PR DESCRIPTION
**Pull request by igor-franca**
Related [ticket](https://liferay.atlassian.net/browse/LPD-26429)

### CODE REVIEW

File: modules/test/playwright/tests/object-web/utils/asyncFilter.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+interface asyncFilter {
+	array: Locator[];
+	predicate: (value: Locator) => Promise<boolean>;
+}
+
+export async function asyncFilter({array, predicate}: asyncFilter) {
+	const results = await Promise.all(array.map(predicate));
+
+	return array.filter((_, index) => results[index]);
+}
```
Comment (danvb112) [inline modules/test/playwright/tests/object-web/utils/asyncFilter.ts:17]: Hey @igor-franca, as we talked personally, I created this [commit](https://github.com/danvb112/liferay-portal/commit/5af0cd144a1cc942cf87e9a4c0463e2bc94f3cc8) by extracting the logic of the async Filter and async Find functions into a generic class, so we can reuse this code in other places as well :smile:

File: modules/test/playwright/tests/object-web/objectFields.spec.ts [hunk 2]
```diff
@@ -43,6 +48,18 @@ test.afterEach(async ({apiHelpers}) => {
...
 	}
 
+	if (createdEntities.objectFolders.length) {
+		await Promise.all(
+			createdEntities.objectFolders.map(async (objectFolder) => {
+				await apiHelpers.objectAdmin.deleteObjectFolder(
+					objectFolder.id
+				);
+			})
+		);
```
Comment (danvb112) [inline modules/test/playwright/tests/object-web/objectFields.spec.ts:58]: Hey @igor-franca, I think we can use the `AsyncArray` class here too, could you try?

**Review by danvb112 — CHANGES_REQUESTED** [review]
Hey @igor-franca, great job refactoring the code! I have some suggestions :smile:

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNodeFields.tsx [hunk 3]
```diff
@@ -51,7 +51,10 @@ export function ObjectDefinitionNodeFields({
 	return (
 		<>
 			{objectFields.map((objectField, index) => {
-				if (index < 5 || showAllObjectFields) {
+				if (
+					objectField.businessType &&
```
[the remark this answers is not in the document]
Comment (igor-franca) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNodeFields.tsx:55]: I completely agree

**Comment by igor-franca** [comment]
I'm experiencing SF errors unrelated to these changes, I'll take a look tomorrow.

**Comment by igor-franca** [comment]
resent [here](https://github.com/liferay-objects/liferay-portal/pull/3632)

**Pull request closed without merging**